aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -27,37 +27,192 @@ Kestrel supports the following features:
+
+ASP.NET Core project templates use Kestrel by default. In *Program.cs*, the template code calls `CreateDefaultBuilder`, which calls [UseKestrel](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.hosting.webhostbuilderkestrelextensions#Microsoft_AspNetCore_Hosting_WebHostBuilderKestrelExtensions_UseKestrel_Microsoft_AspNetCore_Hosting_IWebHostBuilder_) behind the scenes.
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_DefaultBuilder&highlight=7)]
+
+If you use the template code, configure Kestrel options in the `ConfigureServices` method of *Startup.cs*:
+
+[!code-csharp[](kestrel/sample2/Startup.cs?name=snippet_KestrelOptions&highlight=3-4)]
+
+If you don't use `CreateDefaultBuilder`, call `UseKestrel` in *Program.cs* and configure options at the same time:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Main&highlight=13-20)]
+
+For more information about Kestrel settings, see the following classes:
+
+* [KestrelServerOptions](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerOptions.cs)
+* [ListenOptions](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/ListenOptions.cs)
+* [KestrelServerLimits](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerLimits.cs)
+
+
+---
+
+### Kestrel options
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+For information about Kestrel options, see [KestrelServerOptions class](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.server.kestrel.kestrelserveroptions).
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+The Kestrel web server has constraint configuration options that are especially useful in Internet-facing deployments. Here are some of the limits you can set:
+
+- Maximum client connections
+- Maximum request body size
+- Minimum request body data rate
+
+You set these constraints and others in the The `Limits` property of the [KestrelServerOptions](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerOptions.cs) class. The `Limits` property holds an instance of the [KestrelServerLimits](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServerLimits.cs) class. 
+
+**Maximum client connections**
+
+The maximum number of concurrent open HTTP/HTTPS connections can be set for the entire application with the following code:
+
+[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_Limits&highlight=3-4)]
+
+There's a separate limit for connections that have been upgraded from HTTP or HTTPS to another protocol (for example, on a WebSockets request).  After a connection is upgraded, it’s not counted against the `MaxConcurrentConnections` limit anymore.
```

Comment:
* Two-spaces before "After"
* The apostrophe on "it's" is a smart (curly) one. Besides that tho ... change that from "it's not counted" to "it isn't counted"
* I'd drop "anymore" or change it to "any longer"